controllers/slicegateway/slicegateway.go

```diff
@@ -302,23 +302,46 @@ func (r *SliceGwReconciler) deploymentForGatewayServer(g *kubeslicev1beta1.Slice
 }
 
 func (r *SliceGwReconciler) serviceForGateway(g *kubeslicev1beta1.SliceGateway) *corev1.Service {
-	svc := &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "svc-" + g.Name,
-			Namespace: g.Namespace,
-			Labels: map[string]string{
-				"kubeslice.io/slice": g.Spec.SliceName,
+	var svc *corev1.Service
+	if g.Status.Config.SliceGatewayNodePort == 0 {
+		svc = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "svc-" + g.Name,
+				Namespace: g.Namespace,
+				Labels: map[string]string{
+					"kubeslice.io/slice": g.Spec.SliceName,
+				},
 			},
-		},
-		Spec: corev1.ServiceSpec{
-			Type:     "NodePort",
-			Selector: labelsForSliceGwDeployment(g.Name, g.Spec.SliceName),
-			Ports: []corev1.ServicePort{{
-				Port:       11194,
-				Protocol:   corev1.ProtocolUDP,
-				TargetPort: intstr.FromInt(11194),
-			}},
-		},
+			Spec: corev1.ServiceSpec{
+				Type:     "NodePort",
+				Selector: labelsForSliceGwDeployment(g.Name, g.Spec.SliceName),
+				Ports: []corev1.ServicePort{{
+					Port:       11194,
+					Protocol:   corev1.ProtocolUDP,
+					TargetPort: intstr.FromInt(11194),
+				}},
+			},
+		}
+	} else {
+		svc = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "svc-" + g.Name,
+				Namespace: g.Namespace,
+				Labels: map[string]string{
+					"kubeslice.io/slice": g.Spec.SliceName,
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Type:     "NodePort",
+				Selector: labelsForSliceGwDeployment(g.Name, g.Spec.SliceName),
+				Ports: []corev1.ServicePort{{
+					Port:       11194,
+					Protocol:   corev1.ProtocolUDP,
+					TargetPort: intstr.FromInt(11194),
+					NodePort:   int32(g.Status.Config.SliceGatewayNodePort),
+				}},
+			},
+		}
```

Comment:
You don't need the `if` block here. The code in `else` block will cover both the cases.

If `g.Status.Config.SliceGatewayNodePort` is 0, then `NodePort` will be 0 and it will be automatically generated when applied.